### PR TITLE
Switch to prettier for js formatting.

### PR DESCRIPTION
**Pull request by DylanPiercey**
https://github.com/marko-js/marko-prettyprint/issues/20

💃

### CODE REVIEW

File: src/util/formatJS.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    printWidth: printContext.maxLen,
+    singleQuote: printContext.singleQuote,
+    useTabs: printContext.indentString[0] === '\t',
+    tabWidth: printContext.indentString.length
+  };
+
+  if (code.slice(0, 5) === 'class') {
+    // When parsing classes prettier (babylon) requires a class name.
+    // Marko does not, we get around this by parsing as an expression.
+    code = format('(' + code + ');', config).slice(1, -3);
```
Comment (austinkelleher) [inline src/util/formatJS.js]: Should this be `.slice(1, -2)` instead? Where is the extra character coming from?

**Comment by austinkelleher** [comment]
Thanks for doing this @DylanPiercey!

File: src/util/formatJS.js (hunk 1 above)
Comment (DylanPiercey) [reply src/util/formatJS.js]: It adds a newline as well in this case. I updated the PR to make this more clear by instead using `trim`.

**Comment by DylanPiercey** [comment]
@austinkelleher let me know if the `.slice(1, -3)` bit looks better now or what your thoughts are.

**Comment by austinkelleher** [comment]
This looks good to me. There are definitely other prettier options that we could add here. We could technically allow the user to pass a `prettierConfig` option instead, which we'd directly pass to the formatter, but I don't like the idea of locking us down to a single formatter. It would also be harder to use with CLI.

File: src/util/formatJS.js (hunk 1 above)
Comment (austinkelleher) [reply src/util/formatJS.js]: Thanks for clarifying and adding `trim`.

**Pull request merged**